logicalRequirements.md

```diff
@@ -343,6 +343,67 @@ __Additional considerations__
 * A `canShineCharge` object implicitly requires the Speed Booster.
 * A `canShineCharge` object implicitly requires the `canShineCharge` tech if it has more than 0 `shinesparkFrames`.
 
+#### comeInWithRMode object
+A `comeInWithRMode` object represents the need to obtain R-mode when entering the room. It has the following properties:
+* _fromNodes:_ Indicates from what doors this logical requirement expects Samus to enter the room.
+
+__Example:__
+```json
+{"comeInWithRMode": {
+  "fromNodes": [1],
+}}
+```
+
+__Additional considerations__
+
+* A `comeInWithRMode` object implicitly requires X-Ray Scope and a Reserve Tank.
+* A `comeInWithRMode` object implicitly requires the `canEnterRMode` tech.
+* A `comeInWithRMode` requires that one of the indicating nodes in `fromNodes` has a matching `leaveWithDamage`.
+  * The `leaveWithDamage` object must satisfy following requirements in order to match:
```

Comment:
Yeah, another thing is that it's necessary to leave while in a position to be able to use X-Ray on the first frame in the next room, which for horizontal transitions means you have to be on the ground. Maybe we shouldn't try to make this as general and instead name it in a way consistent with it being used for setting up R-mode/G-mode. Maybe something like "leaveWithGModeSetup"? It's a little awkward since R-mode and G-mode have the same setup, but maybe that's ok.